### ISSUE TITLE
Depend on the llvm official lldb-dap extension

### ISSUE THREAD

**Issue by ashgti** [swiftlang/vscode-swift#1246]
Currently vscode-swift depends on `vadimcn.vscode-lldb` but there is an official lldb-dap extension published by the llvm org as `llvm-vs-code-extensions.lldb-dap` see https://marketplace.visualstudio.com/items?itemName=llvm-vs-code-extensions.lldb-dap

Should this depend on that extension instead?

**Comment by plemarquand** [comment]
The `lldb-dap` binary actually ships in the Swift toolchains as of Swift 6.0. You can opt in to it by setting the `swift.debugger.useDebugAdapterFromToolchain` setting to `true`. The long term plan is to phase out the dependency on third party extensions but because the extension supports back to Swift 5.9 we're reliant on it until we can drop support all the way up to 6.0.

I know @matthewbastien looked at the official lldb-dap extension at some point and I remember it didn't fully work for the extension's use cases with older Swift toolchains.

**Comment by matthewbastien** [comment]
The long term plan is to drop support for CodeLLDB and use the `lldb-dap` extension since it's officially supported by LLVM and the executable is shipped with the Swift toolchain. However, it'll require some work in both extensions to be able to co-ordinate between the two.

Since `lldb-dap` was only recently added to the Swift toolchain Paul is correct in that we cannot use it to support projects older than 6.0.

**Comment by plemarquand** [comment]
@matthewbastien thanks, is there an existing issue for that or should we track with this one?

**Comment by adam-fowler** [comment]
Last time I checked lldb-dap still had some stability issues. Has a comparison been done comparing the debugging experience using the two debug adapters.

**Comment by matthewbastien** [comment]
I haven't created the issues for debug yet. So, let's use this one.

**Comment by matthewbastien** [comment]
> Last time I checked lldb-dap still had some stability issues. Has a comparison been done comparing the debugging experience using the two debug adapters.

Yeah, there are definitely still some gaps to investigate. We're going to be working with some folks that help maintain lldb to iron out some of the issues with lldb-dap before we switch over.

**Comment by matthewbastien** [comment]
Fixed by #1384

**Comment by plemarquand** [comment]
Verified with cef4e6cae786dacd1d10a98a6062a50bf96ab112
